lms/services/exceptions.py

```diff
@@ -19,14 +19,16 @@ def __init__(self, explanation=None, response=None, details=None):
         self.details = details
 
     def __str__(self):
+        explanation = self.explanation or "ExternalRequestError"
```

Comment:
I think the string representation should probably be something like:

    "ExternalRequestError(explanation='...', response=Response(status=xxx, reason='...', text='...'))"

That's a lot clearer and more explicit, and can represent cases when either the `explanation` or `Response` is missing:

* `"ExternalRequestError(explanation=None, response=Response(...))"`
* `"ExternalRequestError(explanation='...', response=None)"`
* `"ExternalRequestError(explanation=None, response=None)"`

Might also want to include `self.details` in `__str__()`.

But that's for another PR. For now I've handled the case where `self.explanation` is `None` by defaulting it to `ExternalRequestError` :man_shrugging: